client/src/shared/components/Input/Input.tsx

```diff
@@ -0,0 +1,67 @@
+import { InputHTMLAttributes } from 'react';
+
+import {
+  StyledWrapper,
+  StyledLabel,
+  StyledRequiredMark,
+  StyledInput,
+  StyledHelperText,
+} from './Input.styled';
+
+export type InputProps = {
+  /**
+   * Label text displayed above the input field.
+   * @type {string}
+   */
+  label: string;
+
+  /**
+   * Helper text displayed below the input field.
+   * Useful for showing hints or validation messages.
+   * @type {string}
+   */
+  helperText?: string;
+
+  /**
+   * Whether the input is required.
+   * When true, a red asterisk (*) will be shown next to the label.
+   * This does not trigger browser-native validation unless you also pass it to the DOM via `required`.
+   * @default false
+   */
+  isRequired?: boolean;
+
+  /**
+   * Whether the input is in an error state.
+   * When true, errorMessage will be shown instead of helperText.
+   */
+  error?: boolean;
+
+  /**
+   * Message displayed when the input is invalid.
+   */
+  errorMessage?: string;
+} & InputHTMLAttributes<HTMLInputElement>;
+
+export const Input = ({
+  label,
+  helperText,
+  isRequired = false,
+  error = false,
+  errorMessage,
+  ...props
+}: InputProps) => {
+  return (
+    <StyledWrapper>
+      <StyledLabel htmlFor={props.id || props.name}>
+        {label}
+        {isRequired && <StyledRequiredMark>*</StyledRequiredMark>}
+      </StyledLabel>
+      <StyledInput {...props} error={error} />
```

Comment:
p3. 
error 프롭이 눈에 더 잘 들어오도록 위치를 앞으로 옮겨보는 건 어떤가요??
가독성을 고려해서 다음과 같이 제안드려 봅니다~!
```suggestion
      <StyledInput error={error} {...props} />
```

Reply:
변경했어요!
https://github.com/woowacourse-teams/2025-ah-madda/pull/81/commits/0ab2417177de76aa23740506d75939c6a539fc53